### PR TITLE
Ci/diff

### PR DESCRIPTION
**Pull request by adambutler**
## Description

**tldr;** Adds the required functionality to handle code changes to quickstart repos.

When these repos are pushed to the changes are automatically pulled into Nexmo Developer. If the change has any affect on the output of any Nexmo Developer page then a Pull Request is automatically opened with the changes to the `.repos` directory and in the description a list and of the affected pages and diff of their content changes.

Here is a flow showing how it works:

ci 2f diff

Known issues:

- [x] If a quickstart removes or moves a file required by Nexmo Developer the page will fail to build during the `diff:build:compare` task and exit with a stacktrace & code of `1`.

- [x] Currently, only content under `_documentation` is tested by the diff:
    - [x] Add support for `_tutorials`
    - [x] Add support for `_api`
    - [ ] Add support for `_open_api`

- [x] `X-Hub-Signature` is sent from GitHub to Nexmo Developer on `push`. Ideally, this would only continue to CI if verified.

- [x] A `secret` should be sent from the CI to Nexmo Developer on `POST /jobs/open_pull_request`. Ideally, this would only continue to CI if verified.

- [x] I _think_ the GitHub `push` event is sent for all branches. Ideally, this would only continue to CI if the branch.

- [ ] Currently, we run `rake repo:pull` which pulls all branches when any quickstart has changed. Ideally, this would only pull the repo that has triggered the build.

## Deploy Notes

- [ ] Setup heroku environment:
    - [ ] `TRAVIS_REPO_ID`
    - [ ] `TRAVIS_TOKEN`
    - [ ] `GITHUB_PERSONAL_ACCESS_TOKEN`
    - [ ] `GITHUB_REPO`
    - [ ] `CI_SECRET`

- [ ] Setup webhooks:
    - [ ] `nexmo-node-quickstart`
    - [ ] `nexmo-php-quickstart`
    - [ ] `nexmo-ruby-quickstart`
    - [ ] `nexmo-python-quickstart`
    - [ ] `nexmo-java-quickstart`
    - [ ] `nexmo-dotnet-quickstart`

- [ ] Setup Travis environment:
    - [x] `CI_SECRET`
    - [ ] `CREATE_PULL_REQUEST_ENDPOINT`

### CODE REVIEW

**Comment by adambutler** [comment]
Note: This PR contains the initial changes to bring repos up-to-date.

**Comment by tommorris** [comment]
This is clever. I approve in a general sense.
